tests/playbooks/tests_route_next_hop_ethernet.yml

```
@@ -0,0 +1,148 @@
+    - name: Manage test interface
+      include_tasks: tasks/manage_test_interface.yml
+      vars:
+        state: present
+    - name: Assert device is present
+      include_tasks: tasks/assert_device_present.yml
+    - name: Set type and interface1
+      set_fact:
+        type: "{{ type }}"
+        interface: "{{ interface1 }}"
+    - name: Show interfaces
+      include_tasks: tasks/show_interfaces.yml
+    - name: Manage test interface
+      include_tasks: tasks/manage_test_interface.yml
+      vars:
+        state: present
+    - name: Assert device is present
+      include_tasks: tasks/assert_device_present.yml
+    - name: Run the test
+      block:
+        - name: Configure the IP addresses
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface0 }}"
+                interface_name: "{{ interface0 }}"
+                state: up
+                type: ethernet
+                autoconnect: true
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.3/24
+                    - 2001:db8::2/32
+                  route:
+                    - network: 198.51.10.64
+                      prefix: 26
+                      gateway: 198.51.100.6
+                      metric: 4
+                    - network: 2001:db6::4
+                      prefix: 128
+                      gateway: 2001:db8::1
+                      metric: 2
+              - name: "{{ interface1 }}"
+                interface_name: "{{ interface1 }}"
+                state: up
+                type: ethernet
+                autoconnect: true
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.6/24
+                    - 2001:db8::4/32
+                  route:
+                    - network: 198.51.12.128
+                      prefix: 26
+                      gateway: 198.51.100.1
+                      metric: 2
+        - name: Get the IPv4 routes from the route table main
+          command: ip -4 route
+          register: route_table_main_ipv4
+          ignore_errors: true
+          changed_when: false
+
+        - name: Get the IPv6 routes from the route table main
+          command: ip -6 route
+          register: route_table_main_ipv6
+          ignore_errors: true
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv4
+            routes
+          assert:
+            that:
+              # When running with nm provider, the route will be configured
+              # with `proto static`
+              # In RHEL-6.10 managed host, the attribute in the route is not
+              # equally spaced
+              - route_table_main_ipv4.stdout is search("198.51.10.64/26 via
+                198.51.100.6 dev ethtest0 (proto static)?(\s+)metric 4")
```

```suggestion
                198.51.100.6 dev ethtest0 (proto static )?metric 4")
```
if the output is `198.51.100.6 dev ethtest0 metric 4` then I don't think `search("198.51.10.64/26 via 198.51.100.6 dev ethtest0 (proto static)?(\s+)metric 4")` will match - it will look for "ethtest0" followed by a space followed by one or more spaces which will fail.  If you are concerned that in output there will be more than one space delimiter then you will need to adjust all of your search regexes to use `\s+` instead of ` `.  Otherwise, the suggested search pattern should suffice.